Framework/API/src/AnalysisDataService.cpp

```diff
@@ -167,6 +169,54 @@ void AnalysisDataServiceImpl::remove(const std::string &name) {
   }
 }
 
+/**
+ * @brief random lowercase letter used for generating workspace name in
+ * unique_name and unique_hidden_name
+ * @return Random Char
+ */
+char AnalysisDataServiceImpl::getRandomLowercaseLetter() {
+  static const std::string alphabet = "abcdefghijklmnopqrstuvwxyz";
+  static std::random_device rd;
+  static std::mt19937 gen(rd());
+  static std::uniform_int_distribution<> dis(0, int(alphabet.size() - 1));
+  return alphabet[dis(gen)];
+}
+
+/**
+ * @brief generate a unique name that will not collide with any other workspace name
+ * @param n Size of the sequence (must be a positive number)
+ * @param prefix String to prefix the random name
+ * @param suffix String to suffix the random name
+ * @return String (prefix + n*random char + suffix)
+ */
+const std::string AnalysisDataServiceImpl::uniqueName(const int n, const std::string &prefix,
+                                                      const std::string &suffix) {
+  if (n <= 0) {
+    throw std::invalid_argument("n must be a positive number");
+  }
+  auto randomNameGenerator = [n]() {
+    std::string name;
+    for (int i = 0; i < n; ++i) {
+      name += getRandomLowercaseLetter();
+    }
+    return name;
+  };
+
+  std::string wsName = "";
+  do {
+    wsName = randomNameGenerator();
```

Comment:
what should be compared against the recorded names is `wsName = prefix + randomNameGenerator() + suffix`